olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -63,6 +67,40 @@ TtlForLayer(const std::vector<olp::dataservice::read::model::Layer>& layers,
   return olp::client::ApiError(olp::client::ErrorCode::NotFound,
                                "Layer specified doesn't exist");
 }
+
+read::PartitionsResponse FindPartition(const read::QuadTreeIndex& quad_tree,
+                                       const olp::geo::TileKey& root_tile_key,
+                                       const read::TileRequest& request,
+                                       bool aggregated) {
+  const auto& tile_key = request.GetTileKey();
+
+  // Look for requested tile in the quad tree or its closest ancestor
+  const auto found_index_data = quad_tree.Find(tile_key, aggregated);
+
+  if (!found_index_data) {
+    OLP_SDK_LOG_ERROR_F(kLogTag,
```

Comment:
Rather warning.
Usually, logs should look like this: `FindPartition: tile not found, tile=%s, root=%s, depth=%d, aggregated='%s'`
Doing all these grammatically correct sentences will bring no gain when it crashes :)